src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,28 @@
+import React, { useDebugValue, useEffect, useState } from 'react';
+
+export const useLocalStorage = <S>(
+  key: string,
+  initialState?: S | (() => S),
+): [S, React.Dispatch<React.SetStateAction<S>>] => {
+  const [state, setState] = useState<S>(initialState as S);
+  useDebugValue(state);
+
+  useEffect(() => {
+    const item = localStorage.getItem(key);
+    if (item) setState(parse(item));
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem(key, JSON.stringify(state));
+  }, [state]);
+
+  return [state, setState];
```

Comment:
여기서 state와 setState를 그대로 배열로 넘기는거 보다는 setState를 변경하는 함수 예를들면 setLocalStorage 같은 함수를 만들어서
```suggestion
const setLocalStorage = (value: S) => {
   setState(value);
}
  return {state, setLocalStorage};
```
처럼 객체로 넘겨주면 훨씬 좋을거 같아요! useState의 set을 그대로 넘기는거보다 함수로 한번 래핑해서 넘기는게 더 좋다고 하더라고요! 그리고 배열은 받을때 순서에 의존해서 받게 되니까 배열보다는 객체로 해서 명확하게 변수를 전달하는게 좋을거같습니다!